packages/backend-app-api/src/wiring/BackendInitializer.ts

```diff
@@ -165,6 +165,25 @@ export class BackendInitializer {
       );
       await registerInit.init.func(deps);
     }
+
+    // Once the backend is started, any uncaught errors or unhandled rejections are caught
+    // and logged, in order to avoid crashing the entire backend on local failures.
+    const rootLogger = await this.#serviceHolder.get(
+      coreServices.rootLogger,
+      'root',
+    );
+    process.on('unhandledRejection', (reason: Error) => {
+      (rootLogger?.child({ type: 'unhandledRejection' }) ?? console).error(
```

Comment:
I think console.error gets written to stderr by default, which might not be expected. Personally I would prefer console.info and logger.warn. Perhaps if there is no logger, you can just create one in line and use it?

Reply:
Sounds good.

However I wonder how you feel about the `console.error` going to stderr rather than stdout? Whereas all of the other backstage logs are going to stdout?

Reply:
Yep that's a valid concern, I ended up simply removing the fallback because realistically a root logger will always be available

Reply:
... and also tweaked this to not be used in tests, we want to detect the errors there